rdf4j/src/main/scala/org/w3/banana/rdf4j/Rdf4jOps.scala

```diff
@@ -1,60 +1,60 @@
 import org.w3.banana._
 
 import scala.collection.JavaConverters._
 
-class SesameOps extends RDFOps[Sesame] with SesameMGraphOps with DefaultURIOps[Sesame] {
+class Rdf4jOps extends RDFOps[Rdf4j] with Rdf4jMGraphOps with DefaultURIOps[Rdf4j] {
 
-  val valueFactory: ValueFactory = ValueFactoryImpl.getInstance()
+  val valueFactory: ValueFactory = SimpleValueFactory.getInstance()
 
   // graph
 
-  def emptyGraph: Sesame#Graph = new LinkedHashModel
+  def emptyGraph: Rdf4j#Graph = new LinkedHashModel
 
-  def makeGraph(it: Iterable[Sesame#Triple]): Sesame#Graph = {
+  def makeGraph(it: Iterable[Rdf4j#Triple]): Rdf4j#Graph = {
     val graph = new LinkedHashModel
     it foreach { t => graph add t }
     graph
   }
 
-  def getTriples(graph: Sesame#Graph): Iterable[Sesame#Triple] = graph.asScala
+  def getTriples(graph: Rdf4j#Graph): Iterable[Rdf4j#Triple] = graph.asScala
 
   // triple
 
-  def makeTriple(s: Sesame#Node, p: Sesame#URI, o: Sesame#Node): Sesame#Triple =
+  def makeTriple(s: Rdf4j#Node, p: Rdf4j#URI, o: Rdf4j#Node): Rdf4j#Triple =
     s match {
-      case res:Resource=>  new StatementImpl(res, p, o)
-      case _=> throw new RuntimeException("makeTriple: in Sesame subject " + p.toString + " must be a either URI or BlankNode")
+      case res: Resource => valueFactory.createStatement(res, p, o)
+      case _=> throw new RuntimeException("makeTriple: in RDF4J, subject " + p.toString + " must be a either URI or BlankNode")
     }
 
-  def fromTriple(t: Sesame#Triple): (Sesame#Node, Sesame#URI, Sesame#Node) =  (t.getSubject, t.getPredicate, t.getObject)
+  def fromTriple(t: Rdf4j#Triple): (Rdf4j#Node, Rdf4j#URI, Rdf4j#Node) =  (t.getSubject, t.getPredicate, t.getObject)
 
   // node
 
-  def foldNode[T](node: Sesame#Node)(funURI: Sesame#URI => T, funBNode: Sesame#BNode => T, funLiteral: Sesame#Literal => T): T = node match {
-    case iri: Sesame#URI => funURI(iri)
-    case bnode: Sesame#BNode => funBNode(bnode)
-    case literal: Sesame#Literal => funLiteral(literal)
+  def foldNode[T](node: Rdf4j#Node)(funURI: Rdf4j#URI => T, funBNode: Rdf4j#BNode => T, funLiteral: Rdf4j#Literal => T): T = node match {
+    case iri: Rdf4j#URI => funURI(iri)
+    case bnode: Rdf4j#BNode => funBNode(bnode)
+    case literal: Rdf4j#Literal => funLiteral(literal)
   }
 
   // URI
 
   /**
-   * we provide our own builder for Sesame#URI to relax the constraint "the URI must be absolute"
+   * we provide our own builder for Rdf4j#URI to relax the constraint "the URI must be absolute"
    * this constraint becomes relevant only when you add the URI to a Sesame store
    */
-  def makeUri(iriStr: String): Sesame#URI = {
+  def makeUri(iriStr: String): Rdf4j#URI = {
     try {
-      new URIImpl(iriStr)
+      valueFactory.createIRI(iriStr)
     } catch {
       case iae: IllegalArgumentException =>
-        new URI {
+        new IRI {
           override def equals(o: Any): Boolean = o.isInstanceOf[URI] && o.asInstanceOf[URI].toString == iriStr
```

Comment:
Here you are using the deprecated `URI` of Rdf4J instead of the `Rdf4j#URI` which you set to the recommend `IRI` in the [Rdf4j trait](https://github.com/banana-rdf/banana-rdf/blob/b9001406956251c66f7731f0f4f419a48e06f9ad/rdf4j/src/main/scala/org/w3/banana/rdf4j/Rdf4j.scala#L15-L16)

```scala
 type URI = Rdf4jIRI
 ```

I think that mistake explains many of the compile time warnings.